gdk/common/validate_user_input_recipe.py

```diff
@@ -0,0 +1,84 @@
+import json
+import logging
+import sys
+from pathlib import Path
+
+import gdk.common.consts as consts
+import gdk.common.exceptions.error_messages as error_messages
+import gdk.common.utils as utils
+
+
+def get_recipe():
+    """
+    Loads the component's recipe from the user-input recipe file as a python object.
+
+    Parameters
+    ----------
+        None
+
+    Returns
+    -------
+        recipe_data: component's recipe as a dictionary object if the user-input recipe file is valid.
+    """
+    user_recipe_file = get_user_input_recipe_file()
```

Comment:
GDK parses the recipes without adhering to the case (case-insensitive) https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/main/gdk/commands/component/transformer/BuildRecipeTransformer.py#L25

You may want to add the syntax and semantic validation on the already read file.